runtime/src/tiered_storage/footer.rs

```diff
@@ -0,0 +1,331 @@
+pub const FOOTER_SIZE: i64 = (mem::size_of::<TieredStorageFooter>() + mem::size_of::<u64>()) as i64;
+// The size of the ending part of the footer.  This size should remain unchanged
+// even when the footer's format changes.
+pub const FOOTER_TAIL_SIZE: i64 = 24;
+
+// The ending 8 bytes of a valid tiered account storage file.
+pub const FOOTER_MAGIC_NUMBER: u64 = 0x502A2AB5; // SOLALABS -> SOLANA LABS
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[repr(C)]
+pub struct TieredStorageMagicNumber(pub u64);
+
+impl Default for TieredStorageMagicNumber {
+    fn default() -> Self {
+        Self(FOOTER_MAGIC_NUMBER)
+    }
+}
+
+#[repr(u64)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Deserialize,
+    num_enum::IntoPrimitive,
+    Serialize,
+    num_enum::TryFromPrimitive,
+)]
+#[serde(into = "u64", try_from = "u64")]
+pub enum AccountMetaFormat {
+    #[default]
+    Hot = 0,
+    Cold = 1,
+}
+
+#[repr(u64)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Deserialize,
+    num_enum::IntoPrimitive,
+    Serialize,
+    num_enum::TryFromPrimitive,
+)]
+#[serde(into = "u64", try_from = "u64")]
+pub enum AccountDataBlockFormat {
+    #[default]
+    AlignedRaw = 0,
+    Lz4 = 1,
+}
+
+#[repr(u64)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Deserialize,
+    num_enum::IntoPrimitive,
+    Serialize,
+    num_enum::TryFromPrimitive,
+)]
+#[serde(into = "u64", try_from = "u64")]
+pub enum OwnersBlockFormat {
+    #[default]
+    LocalIndex = 0,
+}
+
+#[repr(u64)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Deserialize,
+    num_enum::IntoPrimitive,
+    Serialize,
+    num_enum::TryFromPrimitive,
+)]
+#[serde(into = "u64", try_from = "u64")]
+pub enum AccountIndexFormat {
+    // This format does not support any fast lookup.
+    // Any query from account hash to account meta requires linear search.
+    #[default]
+    Linear = 0,
+    // Similar to index, but this format also stores the offset of each account
+    // meta in the index block.
+    LinearIndex = 1,
+}
+
+#[derive(Debug)]
+pub struct TieredFileFormat {
+    pub meta_entry_size: usize,
+    pub account_meta_format: AccountMetaFormat,
+    pub owners_block_format: OwnersBlockFormat,
+    pub account_index_format: AccountIndexFormat,
+    pub data_block_format: AccountDataBlockFormat,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+#[repr(C)]
+pub struct TieredStorageFooter {
+    // formats
+    pub account_meta_format: AccountMetaFormat,
+    pub owners_block_format: OwnersBlockFormat,
+    pub account_index_format: AccountIndexFormat,
+    pub data_block_format: AccountDataBlockFormat,
+
+    // account-related
+    pub account_meta_count: u32,
+    pub account_meta_entry_size: u32,
+    pub account_data_block_size: u64,
```

Comment:
They are already together.  Let me change the name to "account_count", "account_entry_size", etc to avoid confusion.

Reply:
Changed `account_meta_count` to `account_entry_count`.  Leave `account_meta_entry_size` unchanged as it really represents account_meta_entry instead of meta + data.

Reply:
> If the data and meta are stored together, I don't understand what account_meta_entry_size could mean then. Can you help me understand, please?

When we store account meta, data, and optional fields together, each account entry consists of account meta, account, and optional fields, as described in the Account Entry section in the proposal:

```
    +-------------------------------+
    | Account Meta                  |
    +-------------------------------+
    | Account Data                  |
    +-------------------------------+
    | Optional Fields               |
    +-------------------------------+
```

To read/parse one account entry given its offset pointing to the first byte of its account entry, we will need to know the size of the account meta entry so that we can obtain the correct offset of the account data, and this is why we need account meta entry size.